easybuild/easyconfigs/e/EasyBuild-custom/EasyBuild-custom-cscs.eb

```diff
@@ -81,12 +81,14 @@ if { ! [ info exists ::env(EASYBUILD_PREFIX) ] } {
 }
 setenv EASYBUILD_INSTALLPATH                    $::env(EASYBUILD_PREFIX)
 #
-# These variables are set to /dev/shm/$USER/easbuild/stage
-# EASYBUILD_BUILDPATH
-# EASYBUILD_TMPDIR
+# If not set, EASYBUILD_BUILDPATH and EASYBUILD_TMPDIR will point to /dev/shm/$USER/easbuild/stage
 #
-setenv EASYBUILD_TMPDIR                         /dev/shm/$::env(USER)/easybuild/stage/tmp
-setenv EASYBUILD_BUILDPATH                      /dev/shm/$::env(USER)/easybuild/stage/build
+if { ! [ info exists ::env(EASYBUILD_TMPDIR) ] } {
+    setenv EASYBUILD_TMPDIR                         /dev/shm/$::env(USER)/easybuild/stage/tmp
+}
```

Comment:
I think you need to add an `else` clause. This is important for the `module unload`. 
Because the `module unload` will rerun the the tcl script and change all the `setenv` to `unset`. But it will never reach the `setenv` without the `else` statement. Because when loaded the `::env(EASYBUILD_TMPDIR)` exists.

Thus, it should look like:
```
if { ! [ info exists ::env(EASYBUILD_TMPDIR) ] } {
    setenv EASYBUILD_TMPDIR                         /dev/shm/$::env(USER)/easybuild/stage/tmp
} else {
    setenv EASYBUILD_TMPDIR                         $::env(EASYBUILD_TMPDIR)
}
```